### PR TITLE
Chore: upgrade flow-bin to latest

### PR DESCRIPTION
**Pull request by halvves**
This is a preparatory PR to get ready for switching to the new context API as mentioned in #1459. `flow-bin` has been upgraded along with any other dependency necessary to keep flow checks running properly. All new warnings/errors surfaced by the upgrade (save
for a few $flowFixMe TODOs) have also been corrected.

I will wait to merge/finalize until #1664 has made its way in (to manage conflicts etc...), but wanted to get this PR up just to see if I was even headed in the right direction as well as get reviews/advice.

R/
halvves

### CODE REVIEW

**Comment by halvves** [comment]
I also meant to ask if we could switch from the manually added flow-typed for `lodash` as that is now throwing an error.

**Comment by probablyup** [comment]
> I also meant to ask if we could switch from the manually added flow-typed for lodash as that is now throwing an error.

Refactoring away from any manual typing when a community-managed version is available seems like the right choice to me.

**Comment by halvves** [comment]
@probablyup - It actually looks like lodash has been removed completely, so I just removed the defs.

**Comment by halvves** [comment]
@probablyup it looks like the appveyor integration is hanging on `eslint` for some reason and erroring out after an hour. `yarn lint` is running fine for me locally and on travis. 😞

**Comment by dashed** [comment]
@halvves looks like `flow-bin` v0.73.0 is now out 😄

**Comment by probablyup** [comment]
@halvves let's do the final upgrade and hopefully close this out. Sorry about AppVeyor... it seems to fail in annoying ways.

**Comment by probablyup** [comment]
@halvves Does this force downstream users to also use 0.73 btw or just for local development? If the former, we'll need to bundle this together with the 4.0 release this summer like the [Typescript one](https://github.com/styled-components/styled-components/pull/1684).

**Comment by halvves** [comment]
@probablyup - My understanding is that this is merely devDependency. It also looks like we are using `rollup-plugin-flow` to strip out flow typings on build, so any change in our inner flow code _shouldn't_ be an issue. I would defer to @mxstbr or @kitten just in case though, since I am pretty new to the proj and may have missed something.

File: .eslintrc [hunk 1]
```diff
@@ -1,20 +1,12 @@
...
-    "prettier/flowtype"
-  ],
+  "extends": ["airbnb", "prettier", "prettier/flowtype"],
   "globals": {
     "__SERVER__": true
   },
   "plugins": ["flowtype", "flowtype-errors", "prettier"],
   "rules": {
-    "semi": 0,
-    "quotes": 0,
```
Comment (probablyup) [inline .eslintrc]: these were all covered by the prettier eslint plugin

File: src/models/StyledComponent.js [hunk 2]
```diff
@@ -178,10 +178,10 @@ export default (ComponentStyle: Function, constructWithOptions: Function) => {
         return
       }
 
-      this.setState(oldState => {
+      this.setState(prevState => {
```
Comment (probablyup) [inline src/models/StyledComponent.js]: the react eslint plugin is kind of dumb and requires you to name stuff like this in a certain way to be detected

**Comment by halvves** [comment]
@probablyup - really sorry about the appveyor stuff... i hope it wasn't something i did 😬

**Comment by probablyup** [comment]
No no nothing on you - it's always flaky :sigh:

**Comment by probablyup** [comment]
Woot!

**Pull request merged**